action.yml

```
@@ -80,13 +83,39 @@ runs:
         version: 'tags/v1.7.2'
         file: 'smack-sint-server-extensions-1.7.2-jar-with-dependencies.jar'
 
+    # When there is an additional jar file for the classpath, merge them into the main jar. I've tried to instead add 
+    # the additional data to the classpath, but that didn't work (although I still believe that that would be the right
+    # approach).
+    - name: Optionally merge extensionJar into test JAR
+      if: ${{ inputs.extensionJar != '' }}
+      run: |
+        set -euo pipefail
+        
+        EXT_JAR="${{ inputs.extensionJar }}"
+        TARGET_JAR="smack-sint-server-extensions-1.7.2-jar-with-dependencies.jar"
+        
+        mkdir -p tmp_extended
+        pushd tmp_extended > /dev/null
+        
+        jar xf "../$EXT_JAR"
+        rm -f META-INF/MANIFEST.MF
+        
+        popd > /dev/null
+        
+        jar uf "$TARGET_JAR" -C tmp_extended .
+        
+        # Remove signature files
+        zip -d "$TARGET_JAR" "META-INF/*.SF" "META-INF/*.RSA" "META-INF/*.DSA" || true
+      shell: bash
+  
     - run: |
           JAVACMD=()
           JAVACMD+=("java")
           JAVACMD+=("-Dsinttest.service=${{ inputs.domain }}")
           JAVACMD+=("-Dsinttest.host=${{ inputs.host }}")
           JAVACMD+=("-Dsinttest.securityMode=disabled")
           JAVACMD+=("-Dsinttest.replyTimeout=${{ inputs.timeout }}")
+          JAVACMD+=("-Dsinttest.testPackages=com,org,net")
```

From a quick skim of Maven Central to collect some common package names.
```suggestion
          JAVACMD+=("-Dsinttest.testPackages=com,org,net,biz,info,dev,io,uk,ch,fr,de,dk,in")
```
Curiosity: Are they called TLDs in this context?

---

Thanks for the additional values. I'd rather have a wildcard, but that's not possible right now. Using commonly used ones for now will minimize the risk of missing out.

And no, I don't think that they're referred to as 'TLD' in this content. A package name using the reverse DNS name is only a convention. I think you'd call them something like 'root of the package name' or something like that. I rarely need a word to describe only that part. :)